pkg/cfn/builder/nodegroup.go

```diff
@@ -59,24 +59,31 @@ func (n *NodeGroupResourceSet) AddAllResources() error {
 	}
 	n.userData = gfn.NewString(userData)
 
+	defaultNodeCount := api.DefaultNodeCount
 	// Ensure MinSize is set, as it is required by the ASG cfn resource
 	if n.spec.MinSize == nil {
 		if n.spec.DesiredCapacity == nil {
-			count := api.DefaultNodeCount
-			n.spec.MinSize = &count
-			logger.Info("--nodes-min=%d was set automatically", count)
+			n.spec.MinSize = &defaultNodeCount
 		} else {
 			n.spec.MinSize = n.spec.DesiredCapacity
 		}
+		logger.Info("--nodes-min=%d was set automatically", *n.spec.MinSize)
 	} else if n.spec.DesiredCapacity != nil && *n.spec.DesiredCapacity < *n.spec.MinSize {
 		return fmt.Errorf("cannot use --nodes-min=%d and --nodes=%d at the same time", *n.spec.MinSize, *n.spec.DesiredCapacity)
 	}
 
 	// Ensure MaxSize is set, as it is required by the ASG cfn resource
 	if n.spec.MaxSize == nil {
 		if n.spec.DesiredCapacity == nil {
-			n.spec.MaxSize = n.spec.MinSize
+			if n.spec.MinSize == nil {
```

Comment:
This if-else seems redundant

Reply:
Yes, because it's actually wrong. Well spotted! 👍